### DISCUSSION TITLE
Start background task from custom api router

### DISCUSSION THREAD

**Discussion by hhertoghe** [fastapi/fastapi#10862] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class APIRouter(FastAPIRouter):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.route_class = LoggingRoute

    def api_route(
        self, path: str, *, include_in_schema: bool = True, **kwargs: Any
    ) -> Callable[[DecoratedCallable], DecoratedCallable]:
        if path.endswith("/"):
            path = path[:-1]

        add_path = super().api_route(
            path, include_in_schema=include_in_schema, **kwargs
        )

        alternate_path = path + "/"
        add_alternate_path = super().api_route(
            alternate_path, include_in_schema=False, **kwargs
        )

        def decorator(func: DecoratedCallable) -> DecoratedCallable:
            add_alternate_path(func)
            return add_path(func)

        return decorator

class LoggingRoute(APIRoute):
    def get_route_handler(self) -> Callable:
        original_route_handler = super().get_route_handler()
        async def custom_route_handler(request: Request) -> Response:
              #send json input body to s3 ... in a background task from here
              response = await original_route_handler(request)
              #send json output body to s3 ... in a background task from here
```

### Description

Hey all,

In my fastapi application, I have a custom API Router that I use to support the trailing slash (code found from the community).
I do have a custom route_class to log information about incoming & outgoing requests.
I send as well to S3 the input & output body processed by my API.
What would be the best way to do those S3 requests in a background task to avoid slowing down the processing of API calls ? I did not find a way to call a background task from here.

Thanks for your help

### Operating System

macOS

### FastAPI Version

0.95.2

### Pydantic Version

1.10.2

### Python Version

3.10

**Comment by jgould22** [comment, marked as answer]
What is the type of `response` in your custom route handler?

You may be able to do it like starlette describes here https://www.starlette.io/background/
